lib/editors.js

```diff
@@ -0,0 +1,1477 @@
+const ERR = require('async-stacktrace');
+const logger = require('../lib/logger');
+const serverJobs = require('../lib/server-jobs');
+const namedLocks = require('../lib/named-locks');
+const syncFromDisk = require('../sync/syncFromDisk');
+const courseUtil = require('../lib/courseUtil');
+const requireFrontend = require('../lib/require-frontend');
+const config = require('../lib/config');
+const path = require('path');
+const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
+const fs = require('fs-extra');
+const async = require('async');
+const uuidv4 = require('uuid/v4');
+const sha256 = require('crypto-js/sha256');
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+function contains(parentPath, childPath) {
+    const relPath = path.relative(parentPath, childPath);
+    return (!(relPath.split(path.sep)[0] == '..' || path.isAbsolute(relPath)));
+}
+
+class Editor {
```

Comment:
This class-based setup with everything in `lib/editor.js` is really nice! Much easier to read!